Networking/NetworkingTests/Remote/ProductVariationsRemoteTests.swift

```diff
@@ -117,6 +117,55 @@ final class ProductVariationsRemoteTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    // MARK: - Load single product variation tests
+
+    /// Verifies that loadProductVariation properly parses the `product-variation` sample response.
+    ///
+    func test_load_single_ProductVariation_properly_returns_parsed_ProductVariation() throws {
+        // Given
+        let remote = ProductVariationsRemote(network: network)
+        let sampleProductVariationID: Int64 = 1275
+        network.simulateResponse(requestUrlSuffix: "products/\(sampleProductID)/variations/\(sampleProductVariationID)", filename: "product-variation")
+
+        // When
+        var resultMaybe: Result<ProductVariation, Error>?
+        waitForExpectation { exp in
+            remote.loadProductVariation(for: sampleSiteID, productID: sampleProductID, variationID: sampleProductVariationID) { aResult in
+                resultMaybe = aResult
+                exp.fulfill()
+            }
+        }
+
+        // Then
+        let result = try XCTUnwrap(resultMaybe)
+        XCTAssertTrue(result.isSuccess)
+
+        let productVariation = try result.get()
+        XCTAssertEqual(productVariation.productID, sampleProductID)
+        XCTAssertEqual(productVariation.productVariationID, sampleProductVariationID)
+    }
+
+    /// Verifies that loadProductVariation properly relays any Networking Layer errors.
+    ///
+    func test_load_single_ProductVariation_properly_relays_netwoking_errors() throws {
+        // Given
+        let remote = ProductVariationsRemote(network: network)
+        let sampleProductVariationID: Int64 = 2783
+
+        // When
+        var resultMaybe: Result<ProductVariation, Error>?
+        waitForExpectation { exp in
+            remote.loadProductVariation(for: sampleSiteID, productID: sampleProductID, variationID: sampleProductVariationID) { aResult in
+                resultMaybe = aResult
+                exp.fulfill()
+            }
+        }
+
+        // Then
+        let result = try XCTUnwrap(resultMaybe)
+        XCTAssertTrue(result.isFailure)
```

Comment:
Do you think we should also add an assertion for the type or message of the error?

Reply:
yup! I added another assertion `XCTAssertEqual(result.failure as? NetworkError, .notFound)` in https://github.com/woocommerce/woocommerce-ios/pull/2915/commits/26e5adeae232736e45eb0e54e0409e259a190c16